specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+
+The developer should always call the Notification Registration API first to register the current process as
+the Activator target.
+
+## Foreground API calls
+
+The developer should always subscribe to AppNotification activation foreground events to intercept notification related actions if
+the app happens to be running in the foreground.
+
+## Manifest Registration
+
+For MSIX, the COM activator GUID and the exe need to be registered in the manifest. The launch args
+would need to be pre-set to a well-known string that defines Activation Triggers.
+
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="----AppNotificationActivationServer">
+        <com:Class Id="{GUIDEntryPoint}" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+# API Details
+
+```c#
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+import "..\AppLifecycle\AppLifecycle.idl";
+
+namespace Microsoft.Windows.AppNotifications
+{
+    // The Registration Info for Packaged and Unpackaged Notification Activations
+    runtimeclass AppNotificationActivationInfo
+    {
+        // Initialize using a manifest defined COM Activator Id. Only applicable to Packaged Win32 applications
+        AppNotificationActivationInfo(Guid taskClsid);
+
+        // Initialize using Notification Assets. Only applicable to Unpackaged Win32 applications which need to specify their own assets like DisplayName and Icon.
+        AppNotificationActivationInfo(String displayName, Windows.Foundation.Uri iconUri);
+
+        // The CLSID associated with the Client COM server that Windows App SDK will activate
+        Guid TaskClsid{ get; };
+
+        // The App friendly DisplayName for the Notification in Action Centre 
+        String DisplayName{ get; };
+
+        // The full file path for the icon image
+        Windows.Foundation.Uri IconUri{ get; };
+    }
+
+    // Event args for the Notification Activation
+    runtimeclass AppNotificationActivatedEventArgs
+    {
+        // Arguments from the invoked button. Empty for Default Activation with no launch args specified in payload.
+        String ActivationArgs{ get; };
+
+        // The data from the input elements of a Notification like a TextBox
+        Windows.Foundation.Collections.IMapView<String, String> UserInput{ get; };
+    }
+
+    // Notification Progress Data
+    runtimeclass AppNotificationProgressData
+    {
+        // Initializes a new Instance of NotificationProgressData
+        AppNotificationProgressData();
+
+        // Gets or sets the sequence number of this notification data.
+        // When multiple NotificationProgressData objects are received, the system displays the data with the greatest non-zero number. 
```

Comment:
Nit: no need to say "non-zero."

Reply:
Good point... which makes this property REQUIRED if the default Uint32 of 0 is used. So unless the ctor of this class sets it to 1 (my preference), it will be a bug farm for developers "why didn't my update show?" "Oh I forgot to put '1' in the stupid sequence field."

Reply:
DISCUSSION: Zero sequence number is special and means "ignore".

Should we make it constructor parameter so you can't forget to set it?

RECOMMEND:
AppNotificationProgressData(UInt32 sequenceNumber);

throw if they pass 0.

Reply:
Done!